modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -902,19 +872,41 @@ public GranuleDescriptor(
         final String granuleLocation = (String) feature.getAttribute(locationAttribute);
         final ReferencedEnvelope granuleBBox = getFeatureBounds(feature);
 
-        // If the granuleDescriptor is not there, dump a message and continue
-        final URL rasterFile = pathType.resolvePath(parentLocation, granuleLocation);
-        if (rasterFile == null) {
+        boolean hasCustomGranuleProvider =
```

Comment:
This block of code (not just the boolean, but also the choice based on the boolean) seems to have commonalities with the other constructor, but not the same. Is there any chance to reduce duplication?

Reply:
I'll check. there are some differences in exception throwing vs null return and url validation in the 2 constructors.

Reply:
Done